pkg/internal/venafi/api/api.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"github.com/Venafi/vcert/pkg/certificate"
+)
+
+// CustomField defines a custom field to be passed to Venafi
+type CustomField struct {
+	Type  certificate.CustomFieldType `json:"type,omitempty"`
```

Comment:
As an end-user, how would I specify a different type? With its integer value?

We should probably define our own `type CustomFieldType string` in this package, and use consts for each type, e.g:

```
type CustomFieldType string
const (
    CustomFieldTypePlain CustomFieldType = "Plain"
)
```

From a user perspective, they'd then specify the following:

```json
{
    "type": "Plain",
    "key": "abc",
    "value": "something
}
```

Which is a lot clearer than `type: 0` 😅

Reply:
I wanted to keep the conversion painless but you're right the typing isn't the best right noe